drivers/SRCommand.py

```diff
@@ -299,6 +299,14 @@ def _run(self, sr, target):
             finally:
                 os.unsetenv('THIN_STATE_FILE_ATTACH')
 
+        elif self.cmd == 'vdi_detach_from_config':
+            extras= {}
+            if target.sr.driver_config.get("ATTACH_FROM_CONFIG_WITH_TAPDISK"):
+                target = blktap2.VDI(self.vdi_uuid, target, self.driver_info)
+                extras['deactivate'] = True
+                extras['caching_params'] = caching_params
+            target.detach(self.params['sr_uuid'], self.vdi_uuid, **extras)
```

Comment:
Shouldn't it be a no-op if the sr driver config doesn't have "ATTACH_FROM_CONFIG_WITH_TAPDISK"?